jaxrs2/src/test/java/feign/jaxrs2/JAXRS2ContractTest.java

```diff
@@ -47,13 +49,57 @@ public void injectJaxrsInternals() throws Exception {
   }
 
   @Test
-  public void injectBeanParam() throws Exception {
+  public void injectWithoutBeanParam() throws Exception {
     final MethodMetadata methodMetadata =
         parseAndValidateMetadata(Jaxrs2Internals.class, "beanParameters", Input.class);
     assertThat(methodMetadata.template())
         .noRequestBody();
   }
 
+  @Test
+  public void injectBeanParam() throws Exception {
+    final MethodMetadata methodMetadata =
+        parseAndValidateMetadata(Jaxrs2BeanParamInternals.class, "beanParameters",
+            BeanParamInput.class);
+    assertThat(methodMetadata.template())
+        .noRequestBody();
+
+    assertThat(methodMetadata.template())
+        .hasHeaders(entry("X-Custom-Header", asList("{X-Custom-Header}")));
+    assertThat(methodMetadata.template())
+        .hasQueries(entry("query", asList("{query}")));
+    assertThat(methodMetadata.formParams())
+        .isNotEmpty()
+        .containsExactly("form");
+
+  }
+
+  public interface Jaxrs2BeanParamInternals {
+    @GET
+    @Path("/")
+    void inject(@Suspended AsyncResponse ar, @Context UriInfo info);
+
+    @Path("/{path}")
+    @POST
+    void beanParameters(@BeanParam BeanParamInput beanParam);
+
+    public class BeanParamInput {
+
+      @PathParam("path")
+      String path;
+
+      @QueryParam("query")
+      String query;
+
+      @FormParam("form")
+      String form;
+
+      @HeaderParam("X-Custom-Header")
+      String header;
+
+    }
+  }
+
 
   @Path("/")
```

Comment:
Now, the compile time error moved to this line.... probably some jaxrs 1 leaking

Reply:
I'll have to install jdk 8 again to check for these issues